crates/sui-core/src/authority.rs

```diff
@@ -687,65 +686,100 @@ impl AuthorityState {
             .handle_node_sync_certificate_latency
             .start_timer();
         let digest = *certificate.digest();
-        debug!(tx_digest = ?digest, "handle_certificate_with_effects");
+        debug!(tx_digest = ?digest, "execute_certificate_with_effects");
         fp_ensure!(
             effects.data().transaction_digest == digest,
             SuiError::ErrorWhileProcessingCertificate {
                 err: "effects/tx digest mismatch".to_string()
             }
         );
 
-        let epoch_store = self.epoch_store();
-        let tx_guard = epoch_store.acquire_tx_guard(certificate).await?;
-
         if certificate.contains_shared_object() {
-            self.database.acquire_shared_locks_from_effects(
-                certificate,
-                effects.data(),
-                &tx_guard,
-            )?;
+            self.database
+                .acquire_shared_locks_from_effects(certificate, effects.data())
+                .await?;
         }
 
-        let resp = self
-            .process_certificate(tx_guard, certificate)
-            .await
-            .tap_err(|e| debug!(?digest, "process_certificate failed: {e}"))?;
-
         let expected_effects_digest = effects.digest();
-        let observed_effects_digest = resp.signed_effects.as_ref().map(|e| e.digest());
-        if observed_effects_digest != Some(expected_effects_digest) {
+
+        let certs = vec![certificate.clone()];
+        self.database
+            .epoch_store()
+            .insert_pending_certificates(&certs)?;
```

Comment:
I don't think we actually need to persist the certificate here - CheckpointExecutor will handle retries if we crash/restart.